### DISCUSSION TITLE
iOS Handling -1005, “the network connection was lost” Errors

### DISCUSSION THREAD

**Discussion by jalopez1986** [MobileNativeFoundation/discussions#96] [General]
Hello, currently where I work, we are having a problem and it is that we are constantly receiving error 1005, “the network connection was lost, the error occurs randomly, it happens with both wifi and 3g and 4g cellular data.

have you had similar problems? How have you solved it?

If the connection is tried again, it works without problem, so I would also like to know what retry strategies do you uses?. Do you code it yourself? Do you use a library?

Currently the connection is made with NSURLConnection

**Comment by vfn** [comment]
I haven't experienced this issue myself, but I have read about it before. [Technical Q&A QA1941](https://developer.apple.com/library/archive/qa/qa1941/_index.html#//apple_ref/doc/uid/DTS40017602) explains the error and suggests approaches for retrying the request. The most important part IMO is:
>Finally, as an app developer it’s reasonable to expect a small number of NSURLErrorNetworkConnectionLost errors caused by problems with the underlying network. However, if you’re seeing a large number of these errors, and especially if they are raised in environments with good network connectivity, there might be a problem with NSURLSession talking to your server. You can investigate that by looking at the on-the-wire traffic to see whether there’s an obvious underlying cause for these errors, for example, your server actually closing connections unexpectedly.

From what you said above...

>...we are constantly receiving error 1005, “the network connection was lost, the error occurs randomly

That should make it "easy" for you to analyse the network traffic to get some more information, as you can easily reproduce it.

>...it happens with both wifi and 3g and 4g cellular data.

If that's also taking into account different devices, where you can rule out the problem as being with a single device, you could infer that the problem is closer to your server or the network infrastructure you have in place, like CDN, firewalls, proxies, &c.

>If the connection is tried again, it works without problem, so I would also like to know what retry strategies do you uses?. Do you code it yourself? Do you use a library?

I'd suggest to first try to identify the problem, and then if needed worry about retry strategies, as they might not be required if the underlying problem is solved. As a generic approach you could always notify the user that an error has occurred giving them the option to retry or cancel it, if that's appropriate for your use case.

Sources:
- https://developer.apple.com/forums/thread/84608?answerId=253347022#253347022
- https://stackoverflow.com/a/39671367/227148
- https://forums.aws.amazon.com/thread.jspa?threadID=119857
